content/en/blog/_posts/2022-04-28-cni-version-related-breaking-change.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: blog
+title: "Breaking changes related to CNI version in popular container runtimes"
+date: 2022-04-28T12:00:00+0000
+slug: cni-version-support-changes-container-runtimes
+---
+
+**Authors:** Mike Brown (IBM), Rolfe Dlugy-Hegwer (Red Hat), and others.
+
+---
+
+_TL;DR: Before you upgrade to Kubernetes 1.24, if you use a CNI network plugin
+that is set up to use Container Network Interface (CNI) versions earlier than
+v1.0.0 and you plan to use any of the following container runtimes, you must
+specify the CNI version. Otherwise, your network will break._
+
+This issue applies to the following container runtimes:
+- **containerd** v1.6.0 and later
+- **CRI-O** 1.24 and later
+
+Here's an overview of the typical steps to perform on each node:
+
+1. [Safely drain and cordon the node](/docs/tasks/administer-cluster/safely-drain-node/).
+2. After stopping your container runtime and kubelet services, perform the
+following upgrade operations:
+  - If you're running CNI plugins, upgrade them to the latest version.
+  - If you're using non-CNI plugins, replace them with CNI plugins. Use the
+  latest version of the plugins.
+  - Update the plugin configuration file to specify or match a version of the
+  CNI specification that the plugin supports.
+  - For `containerd`, ensure that you have installed the latest version (v1.0.0
+  or later) of the CNI loopback plugin.
+  - Upgrade node components (for example, the kubelet) to Kubernetes 1.24
+  - Upgrade to or install the most current version of the container runtime.
+3. Bring the node back into your cluster by restarting your container runtime
+and kubelet. Uncordon the node (`kubectl uncordon <nodename>`).
+
+If you performed the upgrade without following these steps and your network is
+broken, see the following [Troubleshooting](#troubleshooting) section.
+
+## Explanation
+
+CNI enables Kubernetes 1.24 networking. The given container runtimes support the
+most current versions of CNI, v1.0.0 and later. For CNI to work, you must use a
+CNI plugin, and its configuration file must specify the CNI version for which
+the plugin was compiled.
+
+If you use a CNI configuration that does not specify the CNI specification
+version in use, the CNI integration code within the container runtime
+(containerd or CRI-O) assumes that the configuration version for that file is
+v0.1.0, even if this isn't what you intended. As a Kubernetes cluster operator,
+you must review your CNI configuration files to ensure that they specify the
+correct plugin version. If an incorrect version is specified, errors might
+occur.
+
+You must safely drain and cordon a node before upgrading the CNI plugins on the
```

Comment:
All of the CNI-project plugins support in-place upgrades. Users should contact their network vendors to see if they need to drain and cordon.